### PR TITLE
Expanded

### PR DESCRIPTION
**Pull request by davidcarlisle**
add testfile as prototype for tests for new primitives, the test should be usable for all engines that have expandafter (by adding a modified shell script in each engine directory)

### CODE REVIEW

File: texk/web2c/Makefile.in [hunk 1]
```diff
@@ -2420,13 +2420,15 @@ EXTRA_DIST = PROJECTS cftests cpascal.h help.h w2c/config.h \
...
 	$(pdftex_tests) tests/wprob.tex pdftexdir/tests/pdfimage.tex \
+	pdftexdir/tests/exanded.tex\
 	tests/1-4.jpg tests/B.pdf tests/basic.tex \
 	tests/lily-ledger-broken.png $(ttf2afm_tests) \
 	pdftexdir/tests/postV3.afm pdftexdir/tests/postV3.ttf \
 	pdftexdir/tests/postV7.afm pdftexdir/tests/postV7.ttf \
 	$(pdftosrc_tests) pdftexdir/tests/test-13.pdf \
 	pdftexdir/tests/test-13.xref pdftexdir/tests/test-15.pdf \
 	pdftexdir/tests/test-15.xref $(libluasocket_sources) \
+	pdftexdir/tests/exanded.xref\
```
Comment (blefloch) [inline texk/web2c/Makefile.in]: Typo "exanded" ↦ "expanded" here and line 2423.

File: texk/web2c/pdftexdir/tests/expanded.tex [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+\def\typ{\immediate\write10 }
+
+\typ{START}
+
+\show\expanded
+
+\typ{test 1}
+\def\aaa{x}
+\def\bbb{\aaa\aaa}
+\expandafter\def\expandafter\ccc{\expanded{\bbb,\aaa}}
```
Comment (blefloch) [inline texk/web2c/pdftexdir/tests/expanded.tex]: Missing `\expandafter` after `\ccc`.
Also worth having a test like `\expandafter\def\expandafter\ddd\expandafter#\expanded{1{#\number--1}}\show\ddd`.

**Comment by blefloch** [comment]
Off the top of my head, other useful tests:
```
\expanded\relax\space\ifincsname \BOOM\fi{\ifincsname \BOOM\fi}
\showtokens\expandafter{\expanded{#,\noexpand\aaa,\unexpanded{\aaa}}}
\showtokens\expandafter{\expanded{#,\unexpanded{#}}}
\showtokens\expandafter{\expanded\expandafter{\noexpand\aaa}}
\showtokens\expandafter{\expanded\expandafter{\unexpanded{\aaa}}}
\showtokens\expanded{{\expanded{\aaa\noexpand\aaa\noexpand\noexpand\noexpand\aaa}}}
\toks0{\aaa}
\showtokens\expanded{{\the\toks0}}
\expanded\bgroup\show\egroup}
\edef\foo{\expanded{##}}\show\foo
\scantokens{\expanded}
```
(The last one checks that a certain global variable (`def_ref` IIRC) is set correctly.  Compare with `\scantokens{\pdfstrcmp{}}` that produces `File ended while scanning text of ???.`)

**Comment by davidcarlisle** [comment]
@blefloch  thanks I fixed those in my repo and made a second PR

**Pull request merged**